modules/apps/powwow/powwow-portlet/docroot/WEB-INF/src/com/liferay/powwow/meetings/portlet/MeetingsPortlet.java

```diff
@@ -379,6 +396,57 @@ public void updatePowwowMeeting(
 		}
 	}
 
+	protected void checkMaxOccurrence(ResourceRequest resourceRequest, ResourceResponse resourceResponse)
+			throws IOException, InvalidRecurrenceRuleException, PortalException {
+		ServiceContext serviceContext = ServiceContextFactory.getInstance(
+				PowwowMeeting.class.getName(), resourceRequest);
+
+		DateTimeFormatter formatter = DateTimeFormat.forPattern("MM/dd/yyyy");
+		JSONObject jsonResponse = JSONFactoryUtil.createJSONObject();
+		LocalDate maxDateOccurence = new LocalDate();
+
+		TimeZone getTimeZone = getTimeZone(resourceRequest);
+		Calendar startTimeCalendar = getJCalendar(resourceRequest, "startTime", getTimeZone);
+		LocalDate startDate = LocalDate.fromCalendarFields(startTimeCalendar);
+
+		Recurrence recurrence = getRecurrence(resourceRequest);
+
+		String rRule = RecurrenceSerializer.serialize(recurrence);
+
+		try {
+			LocalDateIterable iterator = LocalDateIteratorFactory.createLocalDateIterable(rRule, startDate, true);
+			Iterator<LocalDate> it = iterator.iterator();
+			int i = 50;
+
+			while (i-- > 0 && it.hasNext()) {
+				if (i == 0) {
+					maxDateOccurence = it.next();
+				} else {
+					it.next();
+				}
+			}
+		} catch (ParseException e) {
+			e.printStackTrace();
+		}
+
+		Calendar untilJCalendar = getJCalendar(resourceRequest, "untilDate", getTimeZone);
+		LocalDate untilDayPick = LocalDate.fromCalendarFields(untilJCalendar);
+
+		if(Validator.isNotNull(maxDateOccurence)) {
+			if (untilDayPick.isAfter(maxDateOccurence)) {
+				String maxDate = maxDateOccurence.toString(formatter);
+
+				jsonResponse.put("valid", "false");
+				jsonResponse.put("errorMessage", LanguageUtil.format(serviceContext.getLocale(),
+						"occurrences-exceeded-50-date-should-be-before-or-equal-to-x", maxDate ));
```

Comment:
Let's have the message "Number of occurrences exceeded 50. Please choose a date that's not after X"